Change InMemoryExporter to clear collection before exporting Metrics (#2937)
open-telemetry/opentelemetry-dotnet: docs/metrics/extending-the-sdk/README.md

  does not implement any retry logic.
* Exporters should avoid generating telemetry and causing live-loop, this can be
  done via `OpenTelemetry.SuppressInstrumentationScope`.

> * Exporters receives a batch of `Metric`, and each `Metric` can contain 1 or

Reviewer: ```suggestion
* Exporters receive a batch of `Metric`, and each `Metric` can contain one or
```

>   more `MetricPoint`s. The exporter should perform all actions with the

Reviewer: ```suggestion
  more `MetricPoint`s. The exporter should perform all actions (e.g. serializing etc.) with the
```

>   `Metric`s and `MetricsPoint`s in the Batch before returning control from

Reviewer: ```suggestion
  `Metric`s and `MetricsPoint`s in the batch before returning control from
```

>   `Export`, once the control is returned, the exporter can no longer assume the

Reviewer 2: nit: Rephrase to "exporter can no longer make any assumptions about the state of the Batch..."

* Exporters should avoid generating telemetry and causing live-loop, this can be
  done via `OpenTelemetry.SuppressInstrumentationScope`.
* Exporters receives a batch of `Metric`, and each `Metric`
  can contain 1 or more `MetricPoint`s. After export, users are expected to
  perform all actions with the exported Metrics and MetricPoints. Once control
  is returned, exporter can no longer assume the state of the Batch.

>   (TODO: I need help with the statement here.)

Author: I need help phrasing this correctly.

Author: Should this be "Once control is returned to the exporter, it can no longer assume the state of the Batch."

Reviewer 3: Maybe "the exporter should perform all actions with the Metrics and MetricsPoints in the Batch before returning control from `Export`, once the control is returned, the exporter can no longer assume the state of the Batch."
